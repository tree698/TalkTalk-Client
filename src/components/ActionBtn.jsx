import React from 'react';
import { Link } from 'react-router-dom';
import { BsChevronRight } from 'react-icons/bs';
import Search from './Search';

export default function ActionBtn() {
  const linkStyle =
    'w-[260px] text-xl text-center py-3 border border-superLightGray rounded-md shadow-inner hover:scale-105 hover:shadow-xl transition-all delay-150 duration-300 ease-in-out';
  return (
    <section className="pt-[130px] flex max-w-4xl mx-auto my-14 gap-x-5">
      <Link to="/home/alldrawings" className={`${linkStyle}`}>
        <div className="flex items-center">
          <BsChevronRight className="ml-7 mr-5" />
          <p>All Drawings</p>
        </div>
      </Link>
      <Link to="/home/mydrawings" className={`${linkStyle}`}>
        <div className="flex items-center">
          <BsChevronRight className="ml-7 mr-5" />
          <p>My Drawings</p>
        </div>
      </Link>
      <Search />
    </section>
  );
}
