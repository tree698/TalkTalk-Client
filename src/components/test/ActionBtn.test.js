import { withRouter } from '../../test/utils';
import renderer from 'react-test-renderer';
import ActionBtn from '../ActionBtn';
import { Route } from 'react-router-dom';

describe('Action Btn', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<ActionBtn />} />)
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
