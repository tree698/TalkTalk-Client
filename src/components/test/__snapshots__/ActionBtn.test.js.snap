// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Btn renders correctly 1`] = `
<section
  className="pt-[100px] md:pt-[120px] lg:pt-[140px] px-20 md:px-8 mb-8 md:mb-10 lg:mb-12 grid grid-rows-3 gap-y-1 md:grid md:grid-cols-3 md:grid-rows-1 md:gap-x-3"
>
  <a
    className="text-xs md:text-sm lg:text-base text-center py-1 md:py-2 border border-superLightGray rounded-md shadow-inner hover:scale-105 hover:shadow-xl transition-all delay-150 duration-300 ease-in-out"
    href="/home/alldrawings"
    onClick={[Function]}
  >
    <div
      className="flex items-center justify-center md:justify-start"
    >
      <svg
        className="ml-5 mr-2 md:mr-3 lg:mr-4"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
          fillRule="evenodd"
        />
      </svg>
      <p>
        All Drawings
      </p>
    </div>
  </a>
  <a
    className="text-xs md:text-sm lg:text-base text-center py-1 md:py-2 border border-superLightGray rounded-md shadow-inner hover:scale-105 hover:shadow-xl transition-all delay-150 duration-300 ease-in-out"
    href="/home/mydrawings"
    onClick={[Function]}
  >
    <div
      className="flex items-center justify-center md:justify-start"
    >
      <svg
        className="ml-5 mr-2 md:mr-3 lg:mr-4"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
          fillRule="evenodd"
        />
      </svg>
      <p>
        My Drawings
      </p>
    </div>
  </a>
  <form
    onSubmit={[Function]}
  >
    <div
      className="flex justify-between text-xs md:text-sm lg:text-base py-1 md:py-2 border border-superLightGray rounded-md shadow-inner"
    >
      <input
        className="outline-none placeholder:text-black pl-2 md:pl-3 lg:pl-4"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
      <div
        className="flex items-center pr-1 md:pr-2 lg:pr-3"
      >
        <span
          className="text-xs md:text-sm lg:text-lg hover:scale-125 transition-all delay-150 duration-300 ease-in-out"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 15 15"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <svg
          className="text-xs md:text-sm lg:text-lg text-lightGray"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 15 15"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.5 2C7.77614 2 8 2.22386 8 2.5L8 12.5C8 12.7761 7.77614 13 7.5 13C7.22386 13 7 12.7761 7 12.5L7 2.5C7 2.22386 7.22386 2 7.5 2Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <button
          className="hover:scale-125 transition-all delay-150 duration-300 ease-in-out"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
        </button>
      </div>
    </div>
  </form>
</section>
`;
